screencaptor/src/main/java/com/wealthfront/screencaptor/recyclerviewmutator/RecyclerViewMutator.kt

```diff
@@ -0,0 +1,28 @@
+package com.wealthfront.screencaptor.recyclerviewmutator
+
+import android.view.View
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+abstract class RecyclerViewMutator<S : View, T>(private val viewClass: Class<S>, private val desiredValue: T) {
+  fun mutate(view: S) {
+    mutateView(view, desiredValue)
+  }
+
+  protected abstract fun mutateView(view: S, value: T)
```

Comment:
because of the transient nature of their item views, recyclerviews don't use tags as an undo mechanism. They simply call notifyDataSetChanged when its time to reset items to their original state